src/model.jl

```diff
@@ -22,35 +22,45 @@ function Vars(grid)
 end
 
 function calcN!(N, sol, t, clock, vars, params, grid)
+    
+    dealias!(sol, grid)
 
     vars.uh  .= view(sol, :, :, 1)
     vars.vh  .= view(sol, :, :, 2)
     vars.ϕh  .= view(sol, :, :, 3)
-    vars.Fch .= view(sol, :, :, 4)
 
     @views @. N[:, :, 1] = - im * grid.kr * vars.ϕh                                    # - ∂ϕ/∂x
     @views @. N[:, :, 2] = - im * grid.l  * vars.ϕh                                    # - ∂ϕ/∂y
-    @views @. N[:, :, 3] = - im * (grid.kr * vars.uh + grid.l * vars.vh) * params.c^2 - vars.Fch    # - c^2 * (∂u/∂x - ∂v/∂y) - Fc
-    @views @. N[:, :, 4] = 0                                                           # 0 for now
+    @views @. N[:, :, 3] = - im * (grid.kr * vars.uh + grid.l * vars.vh) * params.c^2  # - c^2 * (∂u/∂x + ∂v/∂y)
 
-    dealias!(N, grid)
+    addforcing!(N, sol, t, clock, vars, params, grid)
 
     return nothing
 
 end
 
+function addforcing!(N, sol, t, clock, vars, params, grid)
+    
+    # call calcF! to compute Fch and store it in vars.Fch
+    params.calcF!(vars.Fch, sol, t, clock, vars, params, grid)
+    
+    # add Fch on the nonlinear term for ϕ
+    @views @. N[:, :, 3] += vars.Fch
+  
+    return nothing
+end
+
 function Equation(params, grid)
 
     T = eltype(grid)
     dev = grid.device
 
-    L = zeros(dev, T, (grid.nkr, grid.nl, 4))
+    L = zeros(dev, T, (grid.nkr, grid.nl, 3))
 
     D = @. - params.ν * grid.Krsq^params.nν  # - ν (k²+l²)ⁿ
 
     @views @. L[:, :, 1] = D - 1 / params.τd    # for u
     @views @. L[:, :, 2] = D - 1 / params.τd    # for v
-    @views @. L[:, :, 3] = D                    # for ϕ
```

Comment:
@natgeo-wong, I removed D from φ since usually the height equation is inviscid... feel free to add it back in